src/mpz.rs

```diff
@@ -158,31 +165,22 @@ impl Mpz {
     //        machinery for a custom type.
     pub fn to_str_radix(&self, base: u8) -> String {
         unsafe {
+            assert!(base >= 2 && base <= 36, "invalid base");
             // Extra two bytes are for possible minus sign and null terminator
-            let len = __gmpz_sizeinbase(&self.mpz, base as c_int) as usize + 2;
+            let len = {
+                let len = __gmpz_sizeinbase(&self.mpz, base as c_int) as usize;
+                assert!(usize::MAX - len >= 2, "capacity overflow");
```

Comment:
This is a nice check, perhaps an error return instead of panic would be nice on the users?

Reply:
This should never happen, so a panic is a better choice.